storage/innobase/trx/trx0rseg.cc

```diff
@@ -101,9 +111,15 @@ trx_rseg_update_wsrep_checkpoint(
 }
 
 /** Update WSREP checkpoint XID in first rollback segment header.
+This function should be called only from innodbase_wsrep_set_checkopoint()
+when it is guaranteed that there are no wsrep transactions committing.
+If the UUID part of the wsrep XID does not match to the UUIDs of XIDs already
+stored into rollback segments, all the remaining XIDs will be reset to empty
+value.
```

Comment:
There is a typo in the function name in the comment. I think that it would be more relevant to mention the function `wsrep_set_SE_checkpoint()`. If that function is invoked at `wsrep_commit()` and other places, then this almost certainly looks like the wrong approach.

The idea of MDEV-15158 was to avoid writing the `TRX_SYS` page during normal operations, and instead have the InnoDB transaction commit write to exactly one of the rollback segment header pages, which is already being written anyway.

With this change, InnoDB would in the worst case update all rollback segment header pages on transaction commit. Granted, it only happens when the UUID part of the XID changes. But, shouldn’t we do this one-time adjustment at InnoDB startup, instead of doing it during normal operation (presumably, on the first transaction commit)?